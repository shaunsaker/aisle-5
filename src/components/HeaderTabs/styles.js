import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingVertical: styleConstants.dimensions.spacing.vertical,
    paddingHorizontal: styleConstants.dimensions.spacing.horizontal,
    borderBottomWidth: 1,
    borderBottomColor: styleConstants.colors.dividerColor,
    height: 51, // item height
  },
  tabContainer: {
    flex: 1,
    alignItems: 'center',
  },
  activeTabContainer: {},
  text: {
    ...styleConstants.fonts.types.small,
    ...styleConstants.fonts.families.bold,
    textDecorationLine: 'underline',
  },
  activeTabText: {
    color: styleConstants.colors.accent,
  },
});

export default styles;
