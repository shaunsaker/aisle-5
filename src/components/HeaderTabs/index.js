import React from 'react';
import PropTypes from 'prop-types';
import { View, Text } from 'react-native';

import styles from './styles';

import Touchable from '../Touchable';

const propTypes = {
  tabs: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string,
    }),
  ).isRequired,
  activeTab: PropTypes.shape({
    text: PropTypes.string,
  }).isRequired,
  handleTabPress: PropTypes.func.isRequired,
  headerTestIDPrefix: PropTypes.string,
};

const defaultProps = {};

const HeaderTabs = ({ tabs, activeTab, handleTabPress, headerTestIDPrefix }) => {
  return (
    <View style={styles.container}>
      {tabs.map((item) => {
        const isActive = item.text === activeTab.text;

        return (
          <Touchable
            key={item.text}
            onPress={() => handleTabPress(item)}
            style={[styles.tabContainer, styles.activeTabContainer]}
            testID={`${headerTestIDPrefix}${item.text}`}
          >
            <Text style={[styles.text, isActive && styles.activeTabText]}>{item.text}</Text>
          </Touchable>
        );
      })}
    </View>
  );
};

HeaderTabs.propTypes = propTypes;
HeaderTabs.defaultProps = defaultProps;

export default HeaderTabs;
