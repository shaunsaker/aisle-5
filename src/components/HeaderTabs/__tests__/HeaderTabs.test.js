import React from 'react';
import renderer from 'react-test-renderer';

import HeaderTabs from '..';

describe('HeaderTabs', () => {
  const tabs = [
    {
      text: 'Foo',
    },
    {
      text: 'Bar',
    },
  ];
  const activeTab = tabs[0];
  const handleTabPress = jest.fn();

  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(
        <HeaderTabs tabs={tabs} activeTab={activeTab} handleTabPress={handleTabPress} />,
      );

      expect(component).toMatchSnapshot();
    });
  });
});
