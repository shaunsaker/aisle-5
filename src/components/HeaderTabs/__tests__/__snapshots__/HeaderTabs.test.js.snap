// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderTabs renders renders with minimum required props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#FFFFFF",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "height": 51,
      "paddingHorizontal": 15,
      "paddingVertical": 10,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "opacity": 1,
      }
    }
    testID="undefinedFoo"
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#212121",
            "fontFamily": "Muli-Bold",
            "fontSize": 15,
            "textDecorationLine": "underline",
          },
          Object {
            "color": "#D17623",
          },
        ]
      }
    >
      Foo
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "opacity": 1,
      }
    }
    testID="undefinedBar"
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#212121",
            "fontFamily": "Muli-Bold",
            "fontSize": 15,
            "textDecorationLine": "underline",
          },
          false,
        ]
      }
    >
      Bar
    </Text>
  </View>
</View>
`;
